File: app/src/main/java/com/example/androidjourney/ui/MainActivity.java
package com.example.androidjourney.ui;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.example.androidjourney.databinding.ActivityMainBinding;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    ArrayList<PostsModel> postsModelArrayList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());


        MyDBHelper myDBHelper = new MyDBHelper(this);

        try {

            JSONArray jsonArray = loadJSONfromAssets();
            Log.d("HELLO"," "+jsonArray);



            for (int i =0; i < jsonArray.length(); i++) {


                JSONObject postData = jsonArray.getJSONObject(i);


                int user_id = postData.getInt("userId");
                String title = postData.getString("title");
                String  body = postData.getString("body");

                myDBHelper.addPosts(user_id,title,body);






            }

        } catch (JSONException e) {
            e.printStackTrace();
        }







    }


    public JSONArray loadJSONfromAssets() {

        JSONArray jsonArray = new JSONArray();

        String json = null;
        try {

            InputStream inputStream = getAssets().open("posts.json");


            int sizeOfFile = inputStream.available();

            byte[] bufferData = new byte[sizeOfFile];

            inputStream.read(bufferData);
            inputStream.close();
            json = new String(bufferData, "UTF-8");
             jsonArray = new JSONArray(json);

        } catch (IOException | JSONException e) {
            e.printStackTrace();

        }
        return jsonArray;

    }
}

File: app/src/main/java/com/example/androidjourney/ui/main/MainActivity.kt
package com.example.androidjourney.ui.main

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.androidjourney.R
import com.example.androidjourney.databinding.ActivityMainBinding
import com.example.androidjourney.ui.fragment.FirstFragment
import com.example.androidjourney.ui.fragment.SecondFragment

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    val Tag = "MMM: First Activity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(Tag, "onCreate")

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar?.hide()

        val secondFragment = SecondFragment()
        val firstFragment = FirstFragment()
        changeFragment(firstFragment)


        binding.btn1.setOnClickListener {
            changeFragment(firstFragment)
        }

        binding.btn2.setOnClickListener {

            changeFragment(secondFragment)
        }
    }


    override fun onStart() {
        super.onStart()
        Log.d(Tag, "onStart")
    }

    override fun onResume() {
        Log.d(Tag, "onResume")
        super.onResume()

    }

    override fun onPause() {
        Log.d(Tag, "onPause")
        super.onPause()

    }

    override fun onStop() {
        Log.d(Tag, "onStop")
        super.onStop()

    }

    override fun onRestart() {
        Log.d(Tag, "onRestart")
        super.onRestart()
    }

    override fun onDestroy() {
        Log.d(Tag, "onDestroy")
        super.onDestroy()
    }

    fun changeFragment(fragment: Fragment) {

        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.main_container, fragment)
        transaction.commit()


    }

}